demos/common/transport/transport_tls_socket_using_mbedtls.c

```diff
@@ -577,7 +577,14 @@ static TlsTransportStatus_t tlsHandshake( NetworkContext_t * pxNetworkContext,
                         lMbedtlsError, mbedtlsHighLevelCodeOrDefault( lMbedtlsError ),
                         mbedtlsLowLevelCodeOrDefault( lMbedtlsError ) ) );
 
-            xRetVal = eTLSTransportHandshakeFailed;
+            if( lMbedtlsError == MBEDTLS_ERR_X509_CERT_VERIFY_FAILED )
+            {
+                xRetVal = eTLSTransportCAVerifyFailed;
```

Comment:
It would be great if we could get more details on why this validation failed.
Ideally the check will verify that the CA root is valid, certificate chaining is valid, no certificate revocation, etc.
(There's a list of other [errors listed here ](https://github.com/Mbed-TLS/mbedtls/blob/273453f126182dd684eb8a5b484ee4c4d2df380c/include/mbedtls/x509.h#L103) but I'm not sure they can be extracted from this API call.

Reply:
The scenario I tried was removing the Baltimore cert from our trusted cert array and leaving Digicert and the MSFT RSA. This is the general error I got which we can use to guide the recovery process. I agree, finer grain details might be nice.